### DISCUSSION TITLE
Load balancers in the mix?

### DISCUSSION THREAD

**Discussion by LMacchi** [choria-io/general#57] [Q&A] [answered]
Hi all,

We are trying to install Choria side-by-side with out Puppet infrastructure. Our Puppet nodes do not talk directly to the compilers, they go through a load balancer:

`Puppet node` -> `load balancer` -> `compiler`

I opened 4222 in the load balancer, and clustered the 3 compilers I'm using to test. However whenever I ran `mco ping`, I get inconsistent results. I suspect the responses get lost in transit?

Example:

I send a `mco ping` from compiler1
node1, node2, node3 receive the request and send the response to the load balancer
load balancer sends responses to compiler1, compiler2, compiler3
compiler1 receives and displays one response

Running the ping several times in a row returns sometimes 3 results, sometimes 1, sometimes 0.

I was expecting that the cluster of compilers would find a way to send the results to compiler1 in order to display them all.

Summing up, my questions:
- Has anyone ever set up Choria with the compilers/borkers behind a load balancer?
- Is my cluster of compilers/brokers behaving as it should or did I set it up incorrectly?

Thanks in advance,

L.

**Comment by ripienaar** [comment, marked as answer]
The communications between the 3 nodes must go direct node-to-node or dedicated load balancer for each IP - but every node must be able to see every other node.

From clients to the cluster you could load balance - though not recommended.

**Reply by LMacchi** [reply]
I see. I appreciate the quick response!
